File: Python/프로그래머스/Level1/[1차] 다트 게임.py
from math import pow
import re

dartResult = '1S*2T*3S'


def solution2(dartResult):
    answer = 0
    return answer


solution2(dartResult)


def solution1(dartResult):
    answer = []
    bonus = {'S': 1, 'D': 2, 'T': 3}
    option = {'#': -1, '*': 2, '': 1}
    p = re.compile('(\d+)([SDT])([*#]?)')
    dart = p.findall(dartResult)
    print(dart)
    for i in range(len(dart)):
        if dart[i][2] == '*':
            if i > 0:
                dart[i-1] *= 2
        dart[i] = int(dart[i][0])**bonus[dart[i][1]]*option[dart[i][2]]
    print(dart)
    return answer

File: Javascript/프로그래머스/Level1/[1차] 다트 게임.js
function solution(dartResult) {
    // 정규표현식 문제
    var answer = 0;
    let result = dartResult.match(/\d+\D[*#]?/g);
    const option = { S: 1, D: 2, T: 3 };
    let scores = [];

    for (let i = 0; i < result.length; i++) {
        let r = result[i];
        let tmp = 0; // SDT의 index저장
        for (let j = 0; j < r.length; j++) {
            if (r[j] === 'S' || r[j] === 'D' || r[j] === 'T') {
                tmp = j;
                break;
            }
        }
        let num = r.slice(0, tmp);

        let score = Number(num) ** option[r[tmp]];
        scores.push(score);

        if (r.includes('*')) {
            scores[i] = scores[i] * 2;
            if (i !== 0) {
                scores[i - 1] = scores[i - 1] * 2;
            }
        }
        if (r.includes('#')) {
            scores[i] = scores[i] * -1;
        }
    }

    scores.map((each) => (answer += each));
    return answer;
}
